types/reply.d.ts

```diff
@@ -12,6 +12,14 @@ export interface ReplyGenericInterface {
   Reply?: ReplyDefault;
 }
 
+type CodeConstrainer<RG extends RouteGenericInterface> =
+  RG['Reply'] extends StatusCodeReply ?
+    HttpCodesCovered<keyof RG['Reply']> : number;
+export type ReplyTypeConstrainer<RG extends RouteGenericInterface, Code extends CodeConstrainer<RG>> =
+  RG['Reply'] extends StatusCodeReply ?
+    RG['Reply'][Code extends keyof RG['Reply'] ? Code : Code extends keyof StatusCodeReply ? CodeToReplyKey<Code> : never] :
+    unknown;
```

Comment:
```ts
export type ReplyTypeConstrainer<RG extends RouteGenericInterface, Code extends CodeConstrainer<RG>> =
  RG['Reply'] extends StatusCodeReply ?
    RG['Reply'][Code extends keyof RG['Reply'] ? Code : Code extends keyof StatusCodeReply ? CodeToReplyKey<Code> : never] :
    unknown;
```

So, this piece of typing does:

1. `RG['Reply'] extends StatusCodeReply` makes it so that `ReplyTypeConstrainer` resolve to `unknown` if `RG['Reply']` is not an object or if the keys of that object are not recognized response codes
2. Else, if `Code` is a key in `RG['Reply']`, then `ReplyTypeConstrainer` will resolve to `RG['Reply'][Code]`
3. Else, if `Code` is a known `ResponseKeys` (why do `extends keyof StatusCodeReply` rather then `extends ResponseKeys`? Just an extra layer of indirection?), then `ReplyTypeConstrainer` will resolve to `RG['Reply'][CodeToReplyKey[Code]]`, where `CodeToReplyKey[Code]`  will equal one of the `1xx` to `5xx` (if `Code` wouldn't be a known response code then `CodeToReplyKey[Code]` would resolve to `Code`, but at this point we know that `Code` is a known `ResponseKeys` , so it will be one of `1xx` to `5xx`). So: `ReplyTypeConstrainer` will resolve to eg. `RG['Reply']['1xx']`
4. Else, if `Code` is neither a key in `RG['Reply']` or a known response key, then `ReplyTypeConstrainer` will resolve to `RG['Reply'][unknown]` (though, due to `1` we should be pretty confident that we won't arrive here)

Purpose of `CodeConstrainer` is to resolve a string literal like `'1xx'` to a number union like `100 | 101 | 102 | 103 | 122`?

---

After having digested this I would propose this implementation instead, together with my other proposed change for `CodeToWildCard` and `WildCardsToCode`:

```suggestion
export type ReplyTypeConstrainer<Reply, Code extends WildCardsToCode<keyof Reply>> =
  Code extends keyof Reply ? Reply[Code] :
    CodeToWildCard<Code> extends keyof Reply ? Reply[CodeToWildCard<Code>] :
      unknown;
```

And this one accepts the `RouteGeneric['Reply']` into it rather than the full interface, as that makes it more readable.